python/Makefile

```
@@ -60,6 +80,142 @@ help:
+	@echo "  1. Review CONTRIBUTING.md for detailed documentation"
+	@echo "  2. Run 'make test' to run tests"
+	@echo "  3. Run 'make help' to see all available commands"
+
+# Quick status check
+.PHONY: status
+status:
+	@# Check development tools
+	@if [ -f .env.versions ]; then \
+		. .env.versions && \
+		if command -v hatch >/dev/null 2>&1; then \
+			HATCH_VER=$$(hatch --version | grep -oE '[0-9]+\.[0-9]+\.[0-9]+'); \
+			if [ "$${HATCH_VER}" = "$${HATCH_VERSION}" ] && python -c "import click; exit(0 if tuple(map(int, click.__version__.split('.'))) < (8, 3, 0) else 1)" 2>/dev/null; then \
+				echo "Development tools: OK"; \
+			else \
+				echo "Development tools: needs update (run 'make setup')"; \
+			fi; \
+		else \
+			echo "Development tools: not installed (run 'make setup')"; \
+		fi; \
+	else \
+		echo "Development tools: config missing"; \
+	fi
+	@# Check Java
+	@if [ -d "$(JAVA_HOME_TARGET)" ]; then \
+		echo "Java 8: OK"; \
+	else \
+		echo "Java 8: not installed"; \
+	fi
+	@# Check virtual environments
+	@if [ -d ".venv" ]; then \
+		COUNT=$$(ls -1 .venv | wc -l | tr -d ' '); \
+		echo "Virtual environments: $${COUNT}"; \
+	else \
+		echo "Virtual environments: none (run 'make venv')"; \
+	fi
+
+# Comprehensive environment health check
+.PHONY: doctor
+doctor:
+	@echo " Running comprehensive environment health check..."
+	@echo "━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━"
+	@echo ""
+	@echo " System Information:"
+	@echo "  OS: $$(uname -s) $$(uname -m)"
+	@echo "  Shell: $$SHELL"
+	@echo "  User: $$USER"
+	@echo ""
+	@echo " Python Status:"
+	@which python >/dev/null 2>&1 && echo "   python: $$(which python) ($$(python --version 2>&1))" || echo "   python: not found"
+	@which python3 >/dev/null 2>&1 && echo "   python3: $$(which python3) ($$(python3 --version 2>&1))" || echo "   python3: not found"
+	@echo ""
+	@$(MAKE) check-deps 2>&1 | sed 's/^/  /'
+	@echo ""
+	@echo " Java Status:"
+	@if [ -d "$(JAVA_HOME_TARGET)" ]; then \
+		echo "   Java 8 installed at $(JAVA_HOME_TARGET)"; \
+	else \
+		echo "    Java 8 not installed (will be installed when needed)"; \
+	fi
+	@echo ""
+	@echo " Virtual Environments:"
+	@if [ -d ".venv" ]; then \
+		for env in .venv/*; do \
+			if [ -d "$$env" ]; then \
+				echo "   $$(basename $$env)"; \
+			fi; \
+		done; \
+	else \
+		echo "    No virtual environments found"; \
+	fi
+	@echo ""
+	@echo " Diagnosis Complete"
+	@if command -v hatch >/dev/null 2>&1 && python -c "import click; exit(0 if tuple(map(int, click.__version__.split('.'))) < (8, 3, 0) else 1)" 2>/dev/null; then \
+		echo "   Environment is healthy"; \
+	else \
+		echo "    Issues detected - run 'make init' to fix"; \
+	fi
+
+# Run all code quality checks (format + lint + type-check)
+.PHONY: quality
+quality: format lint type-check
+	@echo "✅ All quality checks passed!"
+
+
+# Deep clean - remove everything including venvs
+.PHONY: deep-clean
+deep-clean: clean
+	@echo " Deep cleaning - removing all artifacts..."
+	@rm -rf .venv/
```

Can we use the `VENV_DIR` variable here
```suggestion
	@rm -rf ${VENV_DIR}/
```